transitions/registration.js

```diff
@@ -359,23 +359,25 @@ module.exports = {
     });
   },
   assignSchedule: (options, callback) => {
-    getRegistrations(
-      options.db,
-      options.doc.fields && options.doc.fields.patient_id,
-      (err, registrations) => {
-        if (err) {
-          return callback(err);
-        }
-        options.params.forEach(name => {
-          const schedule = schedules.getScheduleConfig(name);
-          const assigned = schedules.assignSchedule(options.doc, schedule, registrations);
-          if (!assigned) {
-            logger.error('Failed to add schedule please verify settings.');
-          }
-        });
-        callback();
+    const patientId = options.doc.fields && options.doc.fields.patient_id;
+
+    async.parallel({
+      registrations: _.partial(getRegistrations, options.db, patientId),
+      person: _.partial(utils.getPatientContact, options.db, patientId)
+    }, (err, {registrations, person}) => {
+      if (err) {
+        return callback(err);
       }
-    );
+      options.params.forEach(name => {
```

Comment:
I'm not sure just reading it what it does either! I'll look into the code and see if I can name it better. The whole interaction with transition config can be really confusing..